### PR TITLE
Fix fedora builds

### PR DESCRIPTION
**Pull request by LecrisUT**
Trying out a build using `dbcsr 2.7.0` to see if that is the cause of recent failures in the Fedora builders

### CODE REVIEW

File: tests/do_regtest.py [hunk 1]
```diff
@@ -246,7 +247,7 @@ async def main() -> None:
 class Config:
     def __init__(self, args: argparse.Namespace):
         self.timeout = args.timeout
-        self.use_mpi = args.version.startswith("p")
+        self.use_mpi = args.mpi or args.version.startswith("p")
```
Comment (LecrisUT) [inline tests/do_regtest.py]: Welp, probably this was the reason why all of the MPI tests were taking so long previously.

File: tests/do_regtest.py [hunk 2]
```diff
@@ -306,11 +307,12 @@ def launch_exe(
             env["CUDA_VISIBLE_DEVICES"] = ",".join(visible_gpu_devices)
             env["HIP_VISIBLE_DEVICES"] = ",".join(visible_gpu_devices)
         env["OMP_NUM_THREADS"] = str(self.ompthreads)
-        exe_path = Path(f"{exe_stem}.{self.version}")
+        exe_name = f"{exe_stem}.{self.version}" if self.version else exe_stem
+        exe_path = Path(exe_name)
```
Comment (oschuett) [inline tests/do_regtest.py]:
```suggestion
        if Path(exe_name).is_absolute():
            cmd = [exe_name]
```
```

File: tests/do_regtest.py [hunk 3]
```diff
@@ -69,7 +70,7 @@ async def main() -> None:
     parser.add_argument("--skip_unittests", action="store_true")
     parser.add_argument("--skip_regtests", action="store_true")
     parser.add_argument("binary_dir", type=Path)
-    parser.add_argument("version")
+    parser.add_argument("version", nargs="?", default=None)
```
Comment (oschuett) [inline tests/do_regtest.py]: Maybe change the default to an empty string as otherwise the `args.version.startswith("p")` below could fail.

**Comment by LecrisUT** [comment]
Ok, got the CI fixed, now just debugging a few other changes to investigate (the 3 `[DEBUG]` commits)

Edit: Finished debugging as well. s390x failures are still relevant (basically the issue is the big-endian-ness), and the LTO failure is :shrug:

**Pull request merged**